src/nvim/ex_cmds.c

```diff
@@ -5399,9 +5402,12 @@ helptags_one (
    * Find all *.txt files.
    */
   dirlen = (int)STRLEN(dir);
-  STRCPY(NameBuff, dir);
-  STRCAT(NameBuff, "/**/*");
-  STRCAT(NameBuff, ext);
+  if (STRLCPY(NameBuff, dir, MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
+  if (STRLCAT(NameBuff, "/**/*", MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
+  if (STRLCAT(NameBuff, ext, MAXPATHL) >= MAXPATHL)
+    EMSG(_(e_pathtoolong));
```

Comment:
This code could potentially print that the path is too long 3x and again continue expanding as if there was no error. It can use just one if statement and a chain of `&&`'s.